docs/features/watchdog.md

```diff
@@ -124,26 +130,174 @@ watchdog.on( 'change:state' ( evt, name, currentState, prevState ) => {
+await watchdog.create( {
+	plugins: []
+} )
+
+// Add editor instances:
+await watchdog.add( [ {
+	id: 'editor1',
+	type: 'editor',
+	sourceElementOrData: document.querySelector( '#editor' ),
+	config: {
+		plugins: [ Essentials, Paragraph, Bold, Italic ],
+		toolbar: [ 'bold', 'italic', 'alignment' ]
+	},
+	creator: ( element, config ) => ClassicEditor.create( element, config )
+}, {
+	id: 'editor2',
+	type: 'editor',
+	sourceElementOrData: document.querySelector( '#editor' ),
+	config: {
+		plugins: [ Essentials, Paragraph, Bold, Italic ],
+		toolbar: [ 'bold', 'italic', 'alignment' ]
+	},
+	creator: ( element, config ) => ClassicEditor.create( element, config )
+} ] );
+```
+
+<info-box>
+	The Watchdog will keep the context and item instances running
+	that are added via the {@link module:watchdog/contextwatchdog~ContextWatchdog#add `ContextWatchdog#add` method}. This method can be called multiple times during the `ContextWatchdog` lifetime.
+
+	To destroy one of the item instances use the {@link module:watchdog/contextwatchdog~ContextWatchdog#remove `ContextWatchdog#remove` method}. This method can be called multiple times during the `ContextWatchdog` lifetime as well.
+</info-box>
+
+```js
+await watchdog.remove( [ 'editor1', 'editor2' ] );
+
+// Or
+await watchdog.remove( 'editor1' );
+await watchdog.remove( 'editor2' );
+```
+
+<info-box>
+	Examples presents the "synchronous way" of the integration with the context watchdog feature, however it's not needed to wait for the promises returned by the `create()`, `add()` and `remove()` methods. There might be a need
+	to create and destroy items dynamically with shared context and that's can be easily achieved as all promises operating on the internal API will be chained.
+</info-box>
+
+#### Context watchdog API
+
+The context watchdog feature provides the following API:
+
+```js
+// Creating watchdog that will use the `Context` class and given configuration.
+const watchdog = new ContextWatchdog( Context, watchdogConfig );
+
+// Setting a custom creator.
+watchdog.setCreator( async config => {
+	const context = await Context.create( config ) );
+
+	// Do something when the context is initialized.
+
+	return context;
+} );
+
+// Setting a custom destructor.
+watchdog.setDestructor( async context => {
+	// Do something before destroy.
+
+	await context.destroy();
+} );
+
+// Initializing the context watchdog with the context configuration.
+await watchdog.create( contextConfig );
+
+// Adding item configuration (or an array of item configurations).
+await watchdog.add( {
+	id: 'editor1',
+	type: 'editor',
+	sourceElementOrData: editorSourceElementOrEditorData
+	config: editorConfig,
+	creator: createEditor,
+	destructor: destroyEditor,
+} );
+
+// Removing and destroy given item.
+await watchdog.remove( [ 'editor1' ] );
+
+// Getting given item instance.
+const editor1 = watchdog.get( 'editor1' );
+
+// Getting given item state.
+const editor1State = watchdog.getState( 'editor1' );
```

Comment:
Having `watchdog.state` property and `watchdog.getState` method is bad already but when they return a different thing, it is horrible. `getItemState`?

Reply:
I'll also change the `watchdog.get( 'editor1' );` to `watchdog.getItem( 'editor1' );`